tools/python/boutiques/puller.py

```diff
@@ -15,43 +16,65 @@
+                                        "zenodo-{0}.json".format(newzid))
+                self.zenodo_entries.append({"zid": newzid, "fname": newfname})
+            except IndexError:
+                raise_error(ZenodoError, "Zenodo ID must be prefixed by "
+                            "'zenodo', e.g. zenodo.123456")
+        self.verbose = verbose
+        self.sandbox = sandbox
+        if(self.verbose):
+            for zid in discarded_zids:
+                print_info("Discarded duplicate id {0}".format(zid))
 
     def pull(self):
         # return cached file if it exists
-        if os.path.isfile(self.cached_fname):
-            if(self.verbose):
-                print_info("Found cached file at %s"
-                           % self.cached_fname)
-            return self.cached_fname
+        json_files = []
+        for entry in self.zenodo_entries:
+            if os.path.isfile(entry["fname"]):
+                if(self.verbose):
+                    print_info("Found cached file at %s"
+                               % entry["fname"])
+                json_files.append(entry["fname"])
+                continue
 
-        from boutiques.searcher import Searcher
-        searcher = Searcher(self.zid, self.verbose, self.sandbox,
-                            exact_match=True)
-        r = searcher.zenodo_search()
+            searcher = Searcher(entry["zid"], self.verbose, self.sandbox,
+                                exact_match=True)
+            r = searcher.zenodo_search()
 
-        for hit in r.json()["hits"]["hits"]:
-            file_path = hit["files"][0]["links"]["self"]
-            file_name = file_path.split(os.sep)[-1]
-            if hit["id"] == int(self.zid):
-                if not os.path.exists(self.cache_dir):
-                    os.makedirs(self.cache_dir)
-                if(self.verbose):
-                    print_info("Downloading descriptor %s"
-                               % file_name)
-                downloaded = urlretrieve(file_path, self.cached_fname)
-                print("Downloaded descriptor to " + downloaded[0])
-                return downloaded[0]
-        raise_error(ZenodoError, "Descriptor not found")
+            if not len(r.json()["hits"]["hits"]):
+                raise_error(ZenodoError, "Descriptor \"{0}\" "
+                            "not found".format(entry["zid"]))
+            for hit in r.json()["hits"]["hits"]:
+                file_path = hit["files"][0]["links"]["self"]
+                file_name = file_path.split(os.sep)[-1]
+                if hit["id"] == int(entry["zid"]):
+                    if not os.path.exists(self.cache_dir):
+                        os.makedirs(self.cache_dir)
+                    if(self.verbose):
+                        print_info("Downloading descriptor %s"
+                                   % file_name)
+                    downloaded = urlretrieve(file_path, entry["fname"])
+                    if(self.verbose):
```

Comment:
I think that this line should be printed even in non-verbose mode. Without it the user has no idea where the descriptor was downloaded and whether the download was successful.

Reply:
Also, you don't need parentheses around `self.verbose` (for this and the one above it)